### PR TITLE
AEA-1189 Fixes to Crypto plugin PR #2112

### PR DESCRIPTION
**Pull request by marcofavorito**
## Proposed changes

This PR includes several improvements wrt #2112 .

- [X] Add version specifiers in contract AEA packages (#2288)
- [x] Add `dependencies:` section in `aea-config.yaml` files (#2289)
- [x] Add relevant crypto dependency in each `aea-config.yaml` in packages (should match default ledger)
- [x] Update demo docs where necessary to include correct dependencies during manual agent construction
- [x] Open issue to write version bumping script
- [X] Add section in `upgrading.md` of how to upgrade (#2303 )

## Fixes

n/a

## Types of changes

What types of changes does your code introduce to agents-aea?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

_Put an `x` in the boxes that apply._

- [ ] I have read the [CONTRIBUTING](../CONTRIBUTING.md) doc
- [ ] I am making a pull request against the `develop` branch (left side). Also you should start your branch off our `develop`.
- [ ] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have checked that code coverage does not decrease.
- [ ] I have checked that the documentation about the `aea cli` tool works
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

## Further comments

n/a

### CODE REVIEW

File: packages/fetchai/contracts/erc1155/contract.yaml [hunk 1]
```diff
@@ -23,8 +23,11 @@ contract_interface_paths:
   ethereum: build/erc1155.json
   fetchai: build/erc1155.wasm
 dependencies:
-  aea_crypto_cosmos: {}
-  aea_crypto_ethereum: {}
-  aea_crypto_fetchai: {}
+  aea_crypto_cosmos:
```
Comment (marcofavorito) [inline packages/fetchai/contracts/erc1155/contract.yaml:26]: this should be `aea-crypto-*`... fixing in incoming PR

File: plugins/aea-crypto-cosmos/setup.py [hunk 2]
```diff
@@ -30,7 +30,7 @@
     license="Apache-2.0",
     description="Python package wrapping the public and private key cryptography and ledger api of Cosmos.",
     packages=find_packages(include=["aea_crypto_cosmos*"]),
-    install_requires=["aea>=0.10.0,<0.11.0", "ecdsa>=0.16", "bech32==1.2.0"],
+    install_requires=["aea>=0.10.0,<0.11.0", "ecdsa>=0.15", "bech32==1.2.0"],
```
Comment (marcofavorito) [inline plugins/aea-crypto-cosmos/setup.py:33]: this was needed for compatibility with PyPI `aea`. Shouldn't hurt, as the latest `ecdsa` available is `0.16.1`, will fix after this release. Opened an issue.

File: packages/fetchai/contracts/erc1155/contract.yaml (hunk 1 above)
Comment (marcofavorito) [reply packages/fetchai/contracts/erc1155/contract.yaml:26]: 0ff7a73

File: packages/fetchai/agents/confirmation_aea_aw1/aea-config.yaml [hunk 3]
```diff
@@ -63,6 +63,9 @@ logging_config:
       level: DEBUG
       propagate: false
 registry_path: ../packages
+dependencies:
```
Comment (DavidMinarsch) [inline packages/fetchai/agents/confirmation_aea_aw1/aea-config.yaml:66]: Missing ethereum

File: tests/data/aea-config.example.yaml [hunk 4]
```diff
@@ -23,4 +23,5 @@ logging_config:
   version: 1
 private_key_paths: {}
 registry_path: ../../packages
-data_dir: .
\ No newline at end of file
+data_dir: .
+dependencies: {}
```
Comment (DavidMinarsch) [inline tests/data/aea-config.example.yaml:27]: any?

Comment (marcofavorito) [reply tests/data/aea-config.example.yaml:27]: why?

**Review by DavidMinarsch — APPROVED** [review]
LGTM

**Pull request merged**